doc/SEMANTICS.md

```diff
@@ -153,19 +153,23 @@ Windows-style path delimiters (`\`) are not supported.
 #### Reads
 
 Basic read-only operations are fully supported, including both sequential and random reads:
-* `open`, `openat`. A file can be opened in read-write mode (`O_RDWR`), but you cannot both read and write to the same file descriptor even in this mode.
+* `open`, `openat`
 * `read`, `readv`, `pread`, `preadv`
 * `lseek`
 * `close`
 
+`open` creates a file handle and returns it back to the kernel. A file handle can only be used for one type of operation, either read or write, for its lifetime. You can open a file in read-write mode (`O_RDWR`), but you cannot both read and write to the same file descriptor even in this mode. The first `read` or `write` will determine the type of operation you can do with the file descriptor.
+
 #### Writes
 
 Mountpoint supports sequential write operations (through `write`, `writev`, `pwrite`, `pwritev`),
 but with some limitations:
 
 * Writes are only supported to new files, and must be done sequentially.
 * Modifying existing files is not supported.
-* Truncation is not supported.
+* Append is not supported, the writes must start from the beginning of the files.
+* Truncation is supported on overwrites.
+* Overwrites can be enabled with `--allow-overwrite` flag and only work when the file is opened in truncate mode (`O_TRUNC`). You cannot overwrite files that are being read, your file descriptor must have exclusive access to the file to start writing. The upload will start as soon as Mountpoint receives the first `write` request and cannot be aborted. Once it is started the file is guaranteed to be overwritten.
```

Comment:
The first bullet is wrong now, and we need to shuffle this around a bit.

Maybe something like:
```
* Writes must start at the beginning of the file and be done sequentially.
* Modifying an existing file is only allowed with the `--allow-overwrite` flag and only when the file is opened in truncate mode (`O_TRUNC`).
    * You cannot overwrite files that are currently being read.
    * The upload to S3 starts as soon as Mountpoint receives the first `write` request and cannot be cancelled.
* Modifying an existing file without using truncate mode is not supported.
```